jwst/saturation/saturation.py

```diff
@@ -160,6 +165,29 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                scigp1 = x_irs2.from_irs2(data[ints, 0, :, :], irs2_mask, detector)
+                mask = scigp1 / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                # Identify groups with suspiciously large values in the second group
+                scigp2 = x_irs2.from_irs2(data[ints, 1, :, :], irs2_mask, detector)
+                mask &= scigp2 > sat_thresh / len(read_pattern[1])
+
+                # Identify groups that are saturated in the third group
+                gp3mask = np.where(flag_temp & SATURATED, True, False)
+                mask &= gp3mask
+
+                # now, flag any pixels that border saturated pixels
+                if n_pix_grow_sat > 0:
+                    mask = adjacency_sat(mask, SATURATED, n_pix_grow_sat)
+
+                # Flag the 2nd group for the pixels passing that gauntlet
+                dq_temp = x_irs2.from_irs2(groupdq[ints, 1, :, :], irs2_mask, detector)
+                dq_temp[mask] |= SATURATED
+                x_irs2.to_irs2(flagarray, dq_temp, irs2_mask, detector)
```

Comment:
I don't think this is being passed back correctly; we're populating flagarray, but flagarray only gets put into the groupdq array a few lines later for group==2 whereas we're trying to set saturation bits for group==1.

Reply:
@drlaw1558 I made some changes and I believe you were correct and now it sets the flags ok, can you verify please?